### PR TITLE
Onboarding

### PR DESCRIPTION
**Pull request by justasm**
#21

Currently this pulls the closest story to the user's location. May want to use a specific story (via remote config) in the future.

untitled drawing-2

### CODE REVIEW

File: app/src/main/java/lt/gyvosistorijos/storage/StoryDb.kt [hunk 1]
```diff
@@ -38,4 +39,14 @@ object StoryDb {
             copyFromRealm(realmStory)?.let { s -> StoryRealm.toStory(s) }
         }
     }
+
+    fun getNearest(lat: Double, lng: Double): Story? {
+        return withRealm {
+            val realmStories = where(StoryRealm::class.java).findAll()
+            copyFromRealm(realmStories)
+                    .map { StoryRealm.toStory(it) }
+                    .sortedBy { distanceMeters(lat, lng, it.latitude, it.longitude) }
```
Comment (vycius) [inline app/src/main/java/lt/gyvosistorijos/storage/StoryDb.kt]: Reuse `StoryDb::getAll`

File: app/src/main/java/lt/gyvosistorijos/ui/onboarding/OnboardingMechanicController.kt [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import com.google.android.gms.maps.model.MarkerOptions
+import kotlinx.android.synthetic.main.controller_onboarding_intro.view.*
+import kotlinx.android.synthetic.main.layout_show_story.view.*
+import lt.gyvosistorijos.*
+import lt.gyvosistorijos.entity.Story
+import lt.gyvosistorijos.location.LocationService
+import lt.gyvosistorijos.storage.StoryDb
+import lt.gyvosistorijos.utils.*
+
+class OnboardingMechanicController : Controller(), LocationListener {
```
Comment (vycius) [inline app/src/main/java/lt/gyvosistorijos/ui/onboarding/OnboardingMechanicController.kt:28]: So many duplicated logic 😢

**Review by vycius — APPROVED** [review]
👍 Onboarding works perfectly.

File: app/src/main/java/lt/gyvosistorijos/ui/onboarding/OnboardingMechanicController.kt (hunk 2 above)
Comment (justasm) [reply app/src/main/java/lt/gyvosistorijos/ui/onboarding/OnboardingMechanicController.kt:28]: Yes 🙂 Had one controller with an enum switch to begin with, but think this approach is clearer / easier to change in the future.

**Pull request merged**